ptrace/debugger/child.py

```diff
@@ -87,6 +88,27 @@ def _createParent(pid, errpipe_read):
         raise child_exception
 
 
+def _closeFdsExcept(fds, ignore_fds):
+    for fd in fds:
+        if fd not in ignore_fds:
+            try:
+                close(fd)
+            except OSError:
+                pass
+
+
+def _closeFds(ignore_fds):
+    for path in ['/proc/self/fd', '/dev/fd']:
```

Comment:
If you want to support BSD, see this test for FreeBSD:
https://github.com/python/cpython/blob/6c3d5274687c97f9c13800ad50e73e15b54f629d/Modules/_posixsubprocess.c#L87